app/javascript/gobierto_dashboards/modules/costs_controller.js

```diff
@@ -115,31 +119,36 @@ export class CostsController {
     }
 
     //Array with all the strings that we've to convert to Number
-    const amountStrings = [ 'cd_bens_i_serveis', 'cd_cost_personal', 'cost_directe' , 'cost_indirecte', 'cost_total', 'costpers', 'costrestadir', 'cost_per_habitant', 'ingressos', 'taxa_o_preu_public', 'cd_serveis_exteriors', 'cd_transferencies', 'cd_equipaments', 'ingres_cost', 'subvencio']
+    const amountStrings = [ 'cost_directe', 'cost_indirecte', 'cost_total', 'cost_per_habitant', 'cost_personal', 'costos_resta_directes', 'ingressos', 'taxa_o_preu_public', 'subvencions', 'ingressos_cost', 'cost_directe_personal', 'cost_directe_bens_i_serveis', 'cost_directe_serveis_exteriors', 'cost_directe_transferencies', 'cost_directe_equipaments', 'cost_directe_financer']
+
+    const population = ['126988']
 
     for (let index = 0; index < rawData.length; index++) {
       let d = rawData[index]
 
+      if (d['any'] === '2019') {
+        d['population'] = population[0]
+      }
+
       for (let amounts = 0; amounts < amountStrings.length; amounts++) {
         d[amountStrings[amounts]] = convertStringToNumbers(d[amountStrings[amounts]])
       }
-
-      d.ingressos = nanToZero(d.ingressos)
     }
 
-    //Split data by year
-    const data2018 = rawData.filter(items => items.any === '2018').map(items => ({ ...items, year: '2018', population: 126988 }))
-    const data2019 = rawData.filter(items => items.any === '2019').map(items => ({ ...items, year: '2019', population: 128265 }))
-
-    const totalData = [...data2018, ...data2019]
+    let yearsCosts = [...new Set(rawData.map(item => item.any))]
 
     let groupDataByYears = []
-    let groupData2018 = []
-    let groupData2019 = []
+
+    //Create an array of objects with all years
+    for (let index = 0; index < yearsCosts.length; index++) {
+      const data = rawData.filter(items => items.any == yearsCosts[index]).map(items => ({ ...items, population: population[index] }))
```

Comment:
`filter` + `map` could be enhanced with `reduce`. Try this:
```suggestion
      const data = rawData.reduce((acc, item) => {
        if (item.any === yearsCosts[index]) {
          acc.push({ ...item, population: population[index] })
        }
        return acc
      }, [])
```
In such way, you run through the array once, whereas filter+map run through it twice
Don't forget the triple comparison